### PR TITLE
feat: Rewrite FP contract distinctions as table

### PR DESCRIPTION
**Pull request by maurelian**
feat: Rewrite FP contract distinctions as table

feat: clarify handling of non-MCP contracts

### CODE REVIEW

**Comment by maurelian** [comment]
* **#11940** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11940?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maurelian and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/scripts/DeployImplementations.s.sol [hunk 1]
```diff
@@ -452,14 +452,16 @@ contract DeployImplementations is Script {
...
-    //   - PreimageOracle: Not proxied, shared by all standard chains.
-    //   - MIPS: Not proxied, shared by all standard chains.
-    //   - OptimismPortal2: Proxied, shared by all standard chains.
+    // | Contract                | Proxied | Deployment                        | MCP Ready  |
+    // |-------------------------|---------|-----------------------------------|------------|
+    // | DisputeGameFactory      | Yes     | Bespoke                           | Yes        |
+    // | AnchorStateRegistry     | Yes     | Bespoke                           | No         |
+    // | FaultDisputeGame        | No      | Bespoke                           | No         |
+    // | PermissionedDisputeGame | No      | Bespoke                           | No         |
+    // | DelayedWETH             | Yes     | Two bespoke (one per DisputeGame) | No         |
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/DeployImplementations.s.sol:461]: DelayedWETH is MCP ready iff all standard chains are required to have the same `DELAY_SECONDS` immutable. This is currently unspecified by the standard config, but I believe we can assume it should be the same. I've asked @pauldowman to get the standard config updated here: https://discord.com/channels/1244729134312198194/1275532049180131429/1285623995789738004

File: packages/contracts-bedrock/scripts/DeployImplementations.s.sol [hunk 2]
```diff
@@ -452,14 +452,16 @@ contract DeployImplementations is Script {
...
+    // | PreimageOracle          | No      | Shared                            | N/A        |
+    // | MIPS                    | No      | Shared                            | N/A        |
+    // | OptimismPortal2         | Yes     | Shared                            | N/A        |
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/DeployImplementations.s.sol]: Similar to above comment, OptimismPortal2 is MCP iff all standard chains are required to share the same values on immutables

**Review by mds1 — APPROVED** [review]
This looks right to me, I like the table format

Let's revisit this pending the standard config update from the proofs team

File: packages/contracts-bedrock/scripts/DeployImplementations.s.sol (hunk 2 above)
Comment (maurelian) [reply packages/contracts-bedrock/scripts/DeployImplementations.s.sol]: OK. This at least shouldn't say `N/A`. That's meant for singletons.

**Pull request merged**